### ISSUE TITLE
Add support for non amd64 platforms.

### ISSUE THREAD

**Issue by ArthurSonzogni** [Crascit/cmake-snap#3]
The snap package is only available on amd64 architecture. It would be nice to have it on every architectures.

See https://forum.snapcraft.io/t/github-to-snap-cmake-version-3-11/12761/12

**Comment by craigscott-crascit** [comment]
I fully agree, but this is a prioritisation issue for me. There are multiple candidates for how to support multiple architectures, but I simply haven't had the spare cycles to explore them.  Unfortunately, I'm not likely to have the time to look into this for a few more months yet.

**Comment by ArthurSonzogni** [comment]
Dump question: wouldn't using "Build with github" be an easy option for you?
[https://snapcraft.io/build](https://snapcraft.io/build)

To unblock me, I just tried doing it myself:
- https://build.snapcraft.io/user/ArthurSonzogni/cmake-build-snap
- https://github.com/ArthurSonzogni/cmake-build-snap

It is working great, but I am not allowed to use the classic confinement. Only you can do it.

**Comment by craigscott-crascit** [comment]
Appreciate the efforts, but I have a responsibility to look into any changes and their implications carefully given the number of people it may affect. There are other aspects like handling tokens securely, etc. which also need to be considered. I'll come back to this when I get a chance, but I'm really not able to look at it for a while.

**Comment by craigscott-crascit** [comment]
@ArthurSonzogni I've done a test run through the recently updated snapcraft build service. There should be a package available for most architectures in the latest/edge channel. Would you be able to test out any of the architectures other than amd64?

FYI I still have some things to sort out and it's not guaranteed I'll be able to ultimately support more than amd64 packages, but let's see how this first set of packages go.

**Comment by ArthurSonzogni** [comment]
Thank you! I will try it on every architectures and post the result back.

**Comment by ArthurSonzogni** [comment]
Thanks you! I am confirming this is working perfectly.
I am now using cmake/latest/edge on 6 projects after those patches:
1. https://github.com/ArthurSonzogni/smk-starter/commit/ec380ef7d4fc28e5e6e9ad0f206ddfcddefbda48
2. https://github.com/ArthurSonzogni/IceMaze/commit/8217399eb5c3b518d5d2feab3bc2d9d75ac4127c
3. https://github.com/ArthurSonzogni/InTheCube/commit/98c8cc25b42256f67dd00d9d9ea28b154af125a3
4. https://github.com/ArthurSonzogni/ftxui-starter/commit/1afaa0fd0b56ac5943fa551802285d078946667e
5. https://github.com/ArthurSonzogni/pigami/commit/3f4927983e1c911b6f653b1b32125d78361f4e0e

This automatically triggered the following snap build on every 6 platforms.
1. https://build.snapcraft.io/user/ArthurSonzogni/smk-starter
2. https://build.snapcraft.io/user/ArthurSonzogni/IceMaze
3. https://build.snapcraft.io/user/ArthurSonzogni/InTheCube
4. https://build.snapcraft.io/user/ArthurSonzogni/ftxui-starter
5. https://build.snapcraft.io/user/ArthurSonzogni/pigami

Everything went correctly on every platform. For instance:
- [s390x](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980081)
- [ppc64el](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980080)
- [arm64](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980079)
- [armhf](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980078)
- [amd64](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980077)
- [i386](https://build.snapcraft.io/user/ArthurSonzogni/IceMaze/980076)

Thanks you!

**Comment by craigscott-crascit** [comment]
Great, thanks for trying that out. I have pushed releases to the beta channel for all six architectures. I'll wait for people to have an opportunity to try them out and confirm they work across the different platforms before pushing them up to stable. To be clear, by "try them out" I'm also referring to running the cmake GUI app on each platform.

**Comment by ArthurSonzogni** [comment]
I don't have any hardware on those platforms, so I won't be able to test cmake GUI app on them myself.

So yes, let's wait on people to try the beta version.

**Comment by craigscott-crascit** [comment]
Rather than pushing to stable, I'll start all architectures with the 3.18 release candidates. Those should start happening in the coming weeks (the release branch has already been created, so it's not far away). I'll consider this issue addressed, the beta builds can be used in the meantime if people don't want to wait.
